rpc/transport/transport.go

```diff
@@ -56,14 +56,36 @@ type Transport interface {
 }
 
 type OutgoingMessage struct {
-	Message rpccp.Message
-	Send    func() error
-	Release capnp.ReleaseFunc
+	Message  rpccp.Message
+	send     func() error
+	released bool
+}
+
+func (o *OutgoingMessage) Release() {
+	if m := o.Message.Message(); !o.released && m != nil {
+		m.Release()
+	}
+}
+
+func (o *OutgoingMessage) Send() error {
+	if !o.released {
+		return o.send()
+	}
+
+	panic("Tried to send() a message that was already released.")
 }
 
 type IncomingMessage struct {
-	Message rpccp.Message
-	Release capnp.ReleaseFunc
+	Message     rpccp.Message
+	ReleaseHook capnp.ReleaseFunc
```

Comment:
This field in particular could use some documentation; really not obvious what it does anymore. More broadly, we need to update the docs for Transport to reflect these changes.